src/scripts/main.js

```diff
@@ -1,3 +1,267 @@
+
+    for (let column = 0; column < 4; column++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function slideUp() {
+  for (let column = 0; column < 4; column++) {
+    let fieldColumn = [board[0][column], board[1][column],
+      board[2][column], board[3][column]];
+
+    fieldColumn = slide(fieldColumn);
+
+    board[0][column] = fieldColumn[0];
+    board[1][column] = fieldColumn[1];
+    board[2][column] = fieldColumn[2];
+    board[3][column] = fieldColumn[3];
+
+    for (let row = 0; row < 4; row++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function slideDown() {
+  for (let column = 0; column < 4; column++) {
+    let fieldColumn = [board[0][column], board[1][column],
+      board[2][column], board[3][column]];
+
+    fieldColumn = slide(fieldColumn.reverse()).reverse();
+
+    board[0][column] = fieldColumn[0];
+    board[1][column] = fieldColumn[1];
+    board[2][column] = fieldColumn[2];
+    board[3][column] = fieldColumn[3];
+
+    for (let row = 0; row < 4; row++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function hasEmptyTiles() {
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 4; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function updateTile(tile, number) {
+  tile.className = '';
+  tile.innerText = '';
+  tile.classList.add('field-cell');
+
+  if (number > 0) {
+    tile.classList.add(`field-cell--${number}`);
+    tile.innerText = number;
+  }
+}
+
+function addNumber() {
+  if (!hasEmptyTiles()) {
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const row = Math.floor(Math.random() * 4);
+    const column = Math.floor(Math.random() * 4);
+
+    if (board[row][column] === 0) {
+      const tile = rows[row].children[column];
+      const randomNuber = newNumber[Math.floor(Math.random() * 10)];
+
+      board[row][column] = randomNuber;
+
+      tile.innerText = randomNuber;
+      tile.classList.add(`field-cell--${randomNuber}`);
+      found = true;
+    }
+  }
+}
+
+function moveIsPossible() {
+  if (hasEmptyTiles()) {
+    return true;
+  }
+
+  let moveX = false;
+  let moveY = false;
+
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 4; c++) {
+      if (board[r][c] === board[r][c + 1]) {
+        moveX = true;
+      }
+    }
+  }
+
+  for (let c = 0; c < 4; c++) {
+    const boardColumn = [board[0][c], board[1][c], board[2][c], board[3][c]];
+
+    for (let r = 0; r < 4; r++) {
+      if (boardColumn[r] === boardColumn[r + 1]) {
+        moveY = true;
+      }
+    }
+  }
+
+  // if (!moveX && !moveY) {
+  //   return false;
+  // }
+
+  // return true;
+  return !((!moveX && !moveY));
```

Comment:
Simplify the return statement in `moveIsPossible` function to `return moveX || moveY;`

Reply:
+